pkg/controller/keystoneservice/keystoneservice_controller.go

```diff
@@ -0,0 +1,163 @@
+)
+
+var log = logf.Log.WithName("controller_keystoneservice")
+
+/**
+* USER ACTION REQUIRED: This is a scaffold file intended for the user to modify with their own Controller
+* business logic.  Delete these comments after modifying this file.*
+ */
+
+// Add creates a new KeystoneService Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileKeystoneService{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("keystoneservice-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource KeystoneService
+	err = c.Watch(&source.Kind{Type: &keystonev1.KeystoneService{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create that are owned by the primary resource
+	// Watch for changes to secondary resource Pods and requeue the owner KeystoneService
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &keystonev1.KeystoneService{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileKeystoneService implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileKeystoneService{}
+
+// ReconcileKeystoneService reconciles a KeystoneService object
+type ReconcileKeystoneService struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a KeystoneService object and makes changes based on the state read
+// and what is in the KeystoneService.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileKeystoneService) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling KeystoneService")
+
+	// Fetch the KeystoneService instance
+	instance := &keystonev1.KeystoneService{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
```

Comment:
I think we'll also want to put something in here to block and Requeue until Keystone bootstrap is completed. Otherwise requests could just fail. During initial deployment we could have multiple services trying to create endpoints, etc at once so blocking until that completes would be ideal. You could look for the BootStrapHash which we are recording in the keystone status objects. If that is set I think it would be safe to continue here as that means the bootstrap job has completed.